src/main/java/org/nibor/autolink/internal/WwwUrlScanner.java

```diff
@@ -0,0 +1,42 @@
+package org.nibor.autolink.internal;
+
+import org.nibor.autolink.LinkSpan;
+import org.nibor.autolink.LinkType;
+
+/**
+ * Scan for URLs starting from the trigger character "w", requires "www.".
+ * <p>
+ * Based on RFC 3986.
+ */
+public class WwwUrlScanner extends UrlScanner {
+
+    @Override
+    public LinkSpan scan(final CharSequence input, int triggerIndex, int rewindIndex) {
+        int afterDot = triggerIndex + 4;
+        if (afterDot >= input.length() || input.charAt(triggerIndex + 1) != 'w' || input.charAt(triggerIndex + 2) != 'w' || input.charAt(triggerIndex + 3) != '.') {
+            return null;
+        }
+
+        int first = triggerIndex;
```

Comment:
That's what I mean, it feels like it shouldn't extract `www.example.org` from `foo.www.example.org`, but with the current code it does:

> org.junit.ComparisonFailure: 
> Expected :foo.www.example.org
> Actual   :foo.|www.example.org|

It even extracts `www.example.org` from `wwww.example.org`. It has to check the characters before the triggerIndex, maybe using `Character.isSpaceChar`.